lib/gollum/app.rb

```diff
@@ -654,9 +667,9 @@ def update_wiki_page(wiki, page, content, commit, name = nil, format = nil)
       wiki.update_page(page, name, format, content.to_s, commit)
     end
 
-    def wiki_page(path, version = nil)
+    def wiki_page(path, version = nil, wiki = nil)
       pathname = (Pathname.new('/') + path).cleanpath
-      wiki = wiki_new
+      wiki = wiki_new if wiki.nil?
```

Comment:
Why not just always do `wiki = wiki_new`? I see that you've added `wiki = nil` as a default parameter to this method, but if I see correctly that's not really doing much, as on https://github.com/gollum/gollum/pull/1759/files#diff-65eee83dd06f55c49a91cd71e716ae5e5ef09b353bf7472d4079ffdc175c395aR443-R444 you basically just pass on `wiki_new` as the parameter.

Reply:
On line 443 I already get a wiki instance using `wiki_new` in order to get a commit object on line 444. But `show_history` method needs the same wiki instance too, so I thought it was not very optimal to create another wiki instance inside the function if it was already created outside of it and can be easily passed to it. If it actually doesn't matter I can always do `wiki = wiki_new`.

Reply:
Thanks for the explanation!